airflow/hooks/druid_hook.py

```diff
@@ -90,3 +107,43 @@ def submit_indexing_job(self, json_index_spec):
                 raise AirflowException('Could not get status of the job, got %s', status)
 
         self.log.info('Successful index')
+
+    def get_first(self, sql, parameters=None):
+        """
+        Executes the druid sql to druid broker and returns the first resulting row.
+
+        :param sql: the sql statement to be executed (str)
+        :type sql: str
+        :param parameters: The parameters to render the SQL query with.
+        :type parameters: mapping or iterable
+        """
+        if sys.version_info[0] < 3:
+            sql = sql.encode('utf-8')
+
+        with closing(self.get_broker_conn()) as conn:
```

Comment:
Curious, does `with self.get_broker_conn() as conn:` not work? We may want to enhance pydruid to implement the context manager (from memory isn't it just about defining an `__enter__` and `__exit__` methods? are they missing?)

Reply:
You are right :) I check pydruid code which has already implements __enter__ and __exit for context manager support. I will update the pr.